src/app/dim-ui/BungieImage.tsx

```diff
@@ -15,8 +18,18 @@ interface BungieImageProps {
 export default function BungieImage(
   props: BungieImageProps & React.ImgHTMLAttributes<HTMLImageElement>
 ) {
-  const { src, ...otherProps } = props;
-  return <img src={bungieNetPath(src)} {...otherProps} />;
+  const { src, hash, ...otherProps } = props;
+
+  let styles = props.className;
+  if (hash) {
+    styles = classNames(props.className, 'perk-image', {
+      'ammo-primary': hash === 143442373,
+      'ammo-special': hash === 2620835322,
+      'ammo-heavy': hash === 2867719094
+    });
```

Comment:
I'd prefer not to add this sort of thing into `BungieImage` - better to make a wrapped component.